diff -up 7zip-25.01/C/LzmaEnc.c.omv~ 7zip-25.01/C/LzmaEnc.c
--- 7zip-25.01/C/LzmaEnc.c.omv~	2025-11-01 20:48:35.899380718 +0100
+++ 7zip-25.01/C/LzmaEnc.c	2025-11-01 20:48:48.674373942 +0100
@@ -2355,7 +2355,7 @@ CLzmaEncHandle LzmaEnc_Create(ISzAllocPt
   p = ISzAlloc_Alloc(alloc, sizeof(CLzmaEnc));
   if (p)
     LzmaEnc_Construct((CLzmaEnc *)p);
-  return p;
+  return (CLzmaEncHandle)p;
 }
 
 static void LzmaEnc_FreeLits(CLzmaEnc *p, ISzAllocPtr alloc)
diff -up 7zip-25.01/C/LzmaEnc.c:.omv~ 7zip-25.01/C/LzmaEnc.c:
diff -up 7zip-25.01/C/Ppmd8.c.omv~ 7zip-25.01/C/Ppmd8.c
--- 7zip-25.01/C/Ppmd8.c.omv~	2025-11-01 20:46:31.416123365 +0100
+++ 7zip-25.01/C/Ppmd8.c	2025-11-01 20:46:57.824896238 +0100
@@ -646,7 +646,7 @@ static CPpmd_Void_Ref CutOff(CPpmd8 *p,
         Ppmd8_InsertNode(p, stats, indx);
       else
         p->UnitsStart += U2B(I2U(indx));
-      stats = ptr;
+      stats = (CPpmd_State*)ptr;
     }
   }
 
diff -up 7zip-25.01/C/XzDec.c.omv~ 7zip-25.01/C/XzDec.c
--- 7zip-25.01/C/XzDec.c.omv~	2025-11-01 20:47:14.774729090 +0100
+++ 7zip-25.01/C/XzDec.c	2025-11-01 20:49:12.638289879 +0100
@@ -279,7 +279,7 @@ SRes Xz_StateCoder_Bc_SetFromMethod_Func
     decoder = (CXzBcFilterState *)ISzAlloc_Alloc(alloc, sizeof(CXzBcFilterState));
     if (!decoder)
       return SZ_ERROR_MEM;
-    decoder->buf = ISzAlloc_Alloc(alloc, BRA_BUF_SIZE);
+    decoder->buf = (Byte*)ISzAlloc_Alloc(alloc, BRA_BUF_SIZE);
     if (!decoder->buf)
     {
       ISzAlloc_Free(alloc, decoder);
@@ -1243,7 +1243,7 @@ SRes XzUnpacker_Code(CXzUnpacker *p, Byt
             UInt32 digest32[XZ_CHECK_SIZE_MAX / 4];
             p->state = XZ_STATE_BLOCK_HEADER;
             p->pos = 0;
-            if (XzCheck_Final(&p->check, (void *)digest32) && memcmp(digest32, p->buf, checkSize) != 0)
+            if (XzCheck_Final(&p->check, (Byte *)digest32) && memcmp(digest32, p->buf, checkSize) != 0)
               return SZ_ERROR_CRC;
             if (p->decodeOnlyOneBlock)
             {
@@ -1292,7 +1292,7 @@ SRes XzUnpacker_Code(CXzUnpacker *p, Byt
             p->state = XZ_STATE_STREAM_INDEX_CRC;
             p->indexSize += 4;
             p->pos = 0;
-            Sha256_Final(&p->sha, (void *)digest32);
+            Sha256_Final(&p->sha, (Byte *)digest32);
             if (memcmp(digest32, p->shaDigest32, SHA256_DIGEST_SIZE) != 0)
               return SZ_ERROR_CRC;
           }
diff -up 7zip-25.01/CPP/7zip/Archive/ComHandler.cpp.omv~ 7zip-25.01/CPP/7zip/Archive/ComHandler.cpp
--- 7zip-25.01/CPP/7zip/Archive/ComHandler.cpp.omv~	2025-11-01 20:49:39.245823420 +0100
+++ 7zip-25.01/CPP/7zip/Archive/ComHandler.cpp	2025-11-01 20:52:46.498956126 +0100
@@ -344,6 +344,7 @@ static UString ConvertName(const Byte *p
   if (CompoundMsiNameToFileName(s, msiName))
   {
     isMsi = true;
+#pragma GCC diagnostic ignored "-Wnrvo"
     return msiName;
   }
   return CompoundNameToFileName(s);
diff -up 7zip-25.01/CPP/7zip/Archive/Udf/UdfIn.cpp.omv~ 7zip-25.01/CPP/7zip/Archive/Udf/UdfIn.cpp
--- 7zip-25.01/CPP/7zip/Archive/Udf/UdfIn.cpp.omv~	2025-11-01 20:52:21.074348607 +0100
+++ 7zip-25.01/CPP/7zip/Archive/Udf/UdfIn.cpp	2025-11-01 20:52:57.919874563 +0100
@@ -118,6 +118,7 @@ static UString ParseDString(const Byte *
     *p = 0;
     res.ReleaseBuf_SetLen((unsigned)(p - (const wchar_t *)res));
   }
+#pragma GCC diagnostic ignored "-Wnrvo"
   return res;
 }
 
diff -up 7zip-25.01/Asm/arm64/7zAsm.S.omv~ 7zip-25.01/Asm/arm64/7zAsm.S
--- 7zip-25.01/Asm/arm64/7zAsm.S.omv~	2025-11-01 20:04:48.710406874 +0000
+++ 7zip-25.01/Asm/arm64/7zAsm.S	2025-11-01 20:05:17.480532677 +0000
@@ -37,6 +37,7 @@
 #define  REG_ABI_PARAM_1 r1
 #define  REG_ABI_PARAM_2 r2
 
+#pragma GCC diagnostic ignored "-Wc++-keyword"
 
 .macro p2_add reg:req, param:req
         add     \reg, \reg, \param
